pallets/clusters/src/lib.rs

```diff
@@ -130,6 +126,19 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type ClustersByOwner<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, Vec<Hash128>>;
 
+	/// **StorageDoubleMap** that maps a (**AccountId**, **Cluster hash**) with a list of **Role** indexes.
+	#[pallet::storage]
+	pub type Members<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, T::AccountId, Identity, Hash128, Vec<u64>>;
+
+	/// **StorageMap** that maps a **(Cluster hash, role name)** to an **index number**
+	#[pallet::storage]
+	pub type ClusterRoleKeys<T: Config> = StorageMap<_, Twox64Concat, (Hash128, Vec<u8>), u64>;
```

Comment:
Isn't this better to be a doublemap though? how is it used?

Reply:
I did not use it because I never need to filter by one of the two keys, which is the main advantage of the `StorageDoubleMap`.

Reply:
If you don't use it we don't need it then?

Reply:
Actually, I could get rid of it, since the index is already included in the `Cluster.roles: BTreeMap<u64, Role>`.
Let me see.

Reply:
No well, I cannot remove it. The idea is:

- Cluster contains its list of roles declared as `roles: BTreeMap<role_index, Role struct>`
- the map between the `Role` in a cluster and its `index` is maintained in `ClusterRoleKeys`, which is a `StorageMap` having a tuple` (cluster_hash, role_name)` as a key. I need a tuple because a role name is unique inside a cluster, but there can be roles with the same name in different clusters
- The role index is a counter kept in `ClusterRoleKeysIndex`, which is a StorageValue. The counter increases every time a new role is created or a new member (with a new role) is added into the cluster.

Basically, `ClusterRoleKeys` and `ClusterRoleKeysIndex` are just utility storages.

Reply:
I'm a bit confused, I will check it better tomorrow but the idea here is to implement it like Tags in protos pallet. Is that the intention?
Basically we want to use some short IDs instead of long names and as well we want to store short IDs instead of full data structures nested.

Reply:
How do I iterate all members of a cluster btw? for example

Reply:
Ok apart from the fact we need to iterate members of a cluster.

I think that the index counter should be inside the cluster; and could be a u16, we don't need more than u16::max roles most likely. Can be u32 to be safe even 🤔

The Account map could just contain clusters.
While the double storage should start with the cluster hash in order to iterate members. This setup goes full circle. As in we can go from account to cluster and deep inside and reverse from cluster to account.

Reply:
Don't you agree that the most common thing users will do is check which are the clusters members? Btw